pkg/controller/eventing/subscription/reconcile.go

```diff
@@ -178,14 +178,24 @@ func (r *reconciler) resolveCall(namespace string, callable v1alpha1.Callable) (
 	//t := duckv1alpha1.Target{}
 	err = duck.FromUnstructured(obj, &t)
 	if err != nil {
-		glog.Warningf("Failed to unserialize legacy target: %s", err)
+		glog.Warningf("Failed to deserialize legacy target: %s", err)
 		return "", err
 	}
 
 	if t.Status.Targetable != nil {
 		return t.Status.Targetable.DomainInternal, nil
 	}
-	return "", fmt.Errorf("status does not contain targetable")
+
+	// K8s Services are special cased.
```

Comment:
If we are going to special case them and need to check for them like so below, I think my preference would be to just fetch them directly as Service objects above Line 171, so see if the reference is to a Service object and if so fetch it as such. Rather than pretend they are duck types.